packages/react-components/src/components/chart/hooks/useDataZoom.ts

```diff
@@ -62,4 +62,111 @@ export const useDataZoom = (chartRef: MutableRefObject<EChartsType | null>, view
     // ignoring because refs dont need to be in dependency array
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [setViewport]);
+
+  // When dragging on the chart, turn off tooltip and set cursor to grabbing
+  const setOnDragStart = (chart: EChartsType | null) => {
```

Comment:
All these events have nothing to do with DataZoom, handle all the user events in a different file. 
We also have handlers listening to these events in the trendcursors file,  if they are dependent, I would move these there, if not, i would suggest a different file something like handleChartEvents

Reply:
Also, all the setoptions happen in one place, so I would return the value using state, i.e the end of the handler would return `{ tooltip: {
        show: false,
      },} ` which then will be used in the last hook which sets the options.

Reply:
Updated! Didn't know `setOptions` is only set in one place, good callout. Moved everything to its own file.